dsgrid/config/dimensions.py

```diff
@@ -49,6 +49,29 @@ class DimensionBaseModel(DSGBaseModel):
         ),
         updateable=False,
     )
+    display_name: str = Field(
+        title="display_name",
+        description="Display name. Source for auto-generated query_name.",
+        note="Dimension display names should be singular noun phrases that are concise and "
+        "distinguish the dimension across all dimensions within a project, inclusive of dataset "
+        "dimensions, project base dimensions, and project supplemental dimensions. This uniqueness "
+        "requirement applies unless the dimension is trivial, that is, contains only a single "
+        "record. For trivial dimensions, if the dimension represents an aggregation of everything, "
+        "e.g., all of the sectors, then the convention is to list the display name as 'None'; "
+        "if the dimension represents a single slice of the data, e.g., all electricity use or "
```

Comment:
I'm OK with the options listed in the PR comments: only require unique display names within a dimension, skip the check for trivial dimensions, allow dashes.

I think I'm still hung up on one item that may be important for the queries. When I look at the SS project.toml I see that end uses are registered as a dimension. When I look inside the enduses_kw.csv I see that one column is `fuel_id`. One typical value is `electricity`. As a software developer this tells me that if I want to find all `electricity` records I should read this table and filter it. But then I see that there is a supplemental dimension called `all-electric-end-uses`. The mapping record file for that dimension seems to have all of the rows in the previous table with `fuel_id = electricity` (at least I hope it does).

Why do we have what are essentially duplicate mappings? Are we checking to ensure they match? Which one do we use for queries? Should we auto-generate the second table during registration? 

This same idea applies to counties and states. Maybe others, too.

Reply:
I like the idea of auto-generating the second table during registration. That would support (a) the dimension records we have now that are convenient for users that want to concisely communicate mapping information, and (b) the ability to use these sorts of auto-generated mappings in the same way as we use explicitly provided mappings.

Reply:
Let's have a discussion on how dimension record fields are to be used generally. It's possible that we can use them to auto-generate mappings, but from chatting with @lixiliu, it doesn't seem likely.

What we definitely could do is add an additional section to `ProjectConfigModel` called something like `all_in_one_supplemental_dimensions`. Instead of a bunch of dimensions and mappings like this:
```
[[dimensions.supplemental_dimensions]]
class = "Sector"
description = """
All Sectors
"""
file = "dimensions/supplemental/all_sectors.csv"
module = "dsgrid.dimension.standard"
name = "all-sectors"
display_name = "All Sector"
type = "sector"
```
```
[[dimension_mappings.base_to_supplemental]]
description = "Aggregation map for Com-Res-Tra sectors"
file = "dimension_mappings/base_to_supplemental/lookup_sectors_to_all_sectors.csv"
mapping_type = "many_to_one_aggregation"
[dimension_mappings.base_to_supplemental.from_dimension]
name = "Sectors - Com-Res-Tra-Only"
type = "sector"
[dimension_mappings.base_to_supplemental.to_dimension]
name = "all-sectors"
type = "sector"
```
we have this one field
```
[dimensions.all_in_one_supplemental_dimensions]
types = ["sector", "subsector", "data_source"]
```
When the registration code sees this entry it auto-generates the dimensions and mappings that are currently generated.